table/block_based/block_based_table_reader.cc

```diff
@@ -2319,30 +2320,71 @@ void BlockBasedTable::RetrieveMultipleBlocks(
   autovector<ReadRequest, MultiGetContext::MAX_BATCH_SIZE> read_reqs;
   size_t buf_offset = 0;
   size_t idx_in_batch = 0;
+
+  uint64_t prev_offset = 0;
+  size_t prev_len = 0;
+  autovector<size_t, MultiGetContext::MAX_BATCH_SIZE> req_idx_for_block;
+  autovector<size_t, MultiGetContext::MAX_BATCH_SIZE> req_offset_for_block;
   for (auto mget_iter = batch->begin(); mget_iter != batch->end();
        ++mget_iter, ++idx_in_batch) {
     const BlockHandle& handle = (*handles)[idx_in_batch];
     if (handle.IsNull()) {
       continue;
     }
 
+    size_t prev_end = static_cast<size_t>(prev_offset) + prev_len;
+
+    // If current block is adjacent to the previous one, at the same time,
+    // compression is enabled and there is no compressed cache, we combine
+    // the two block read as one.
+    if (rep_->blocks_maybe_compressed &&
```

Comment:
Can we change this to ```scratch != nullptr && prev_end == handle.offset()```? Since the caller would have already checked, let's eliminate the code duplication and the risk of the two getting out of sync.

Reply:
@anand1976  I will correct them before landing. Thanks for reviewing the PR!